### ISSUE TITLE
Dualminer USB in scrypt mode

### ISSUE THREAD

**Issue by wareck** [luke-jr/bfgminer#573]
Hi
I've got some trouble
I'm trying to use Dualminer USB in scrypt mode only
And bfgminer seem's to not works
After one hour hashrate is still very slow
Leds on Dualminer flicking like "connected" "deconnected"
and I have this:
![dmu_fail](https://cloud.githubusercontent.com/assets/5462099/4902752/9229e0ce-643b-11e4-8170-2c32bf755042.jpg)

In some case dmu starts in SHA256 so bfgminer sends work hash to pool server and 100% are rejected...

I know these USB dualminer are old but I've got lot of them...

**Comment by luke-jr** [comment]
Are you sure the USB hub has enough power for them all?

**Comment by wareck** [comment]
Yes because with cgminer they works since 3 weeks without issue.
It's only in scrypt mode, in sha256 works greats

**Comment by wareck** [comment]
Ok I tries some thinks...

First if I remove everything about Dualmode or SHA mode in the last driver-dualminer.c
=> Dualminer works in scypt mode, but don't share at is optimal speed (half, 35kh/s instead of 70)

It's because dualminer restart it's init every 5/6 seconds

So I think there is something wrong into:
-void dualminer_init_firstrun (don't init or reset init every hash send)
-and something wrong into switch/mode selection (SHA,SCRYPT or DUAL)

I'll try to help you and us to find where is the problem...
I have some skill in C, but I'm searching a good IDE on linux to test my code.
Can you advice me , did you use on IDE for build bfgminer ?

Olivier

**Comment by luke-jr** [comment]
No, I just use a text editor.

**Comment by wareck** [comment]
Ok...
I'm searching :)
I see that there is many "ICARUS" into dualminer code
Driver-dualminer.c is independant or works with icarus ?
(edit: ok...Icarus.common :) )
